File: sdk/ocaml/lib/dns_manage.ml
(** DNS Manage: domain DNS records, DNSSEC, nameserver registration *)

(** GET /api/domain/{id}/dns - List DNS records *)
let get_records client params =
  HttpClient.get client "/api/domain/{id}/dns" params

(** POST /api/domain/{id}/dns - Create DNS record *)
let create_record client params =
  HttpClient.post client "/api/domain/{id}/dns" params

(** PUT /api/domain/{id}/dns/{index} - Update DNS record *)
let update_record client params =
  HttpClient.put client "/api/domain/{id}/dns/{index}" params

(** DELETE /api/domain/{id}/dns/{index} - Remove DNS record *)
let delete_record client params =
  HttpClient.delete client "/api/domain/{id}/dns/{index}" params

(** GET /api/domain/{id}/dns/types - List supported record types *)
let get_record_types client params =
  HttpClient.get client "/api/domain/{id}/dns/types" params

(** GET /api/domain/{id}/dnssec/flags - Get available DNSSEC flags *)
let get_dnssec_flags client params =
  HttpClient.get client "/api/domain/{id}/dnssec/flags" params

(** GET /api/domain/{id}/dnssec - Get DNSSEC records *)
let get_dnssec_records client params =
  HttpClient.get client "/api/domain/{id}/dnssec" params

(** POST /api/domain/{id}/dnssec - Create DNSSEC record *)
let create_dnssec_record client params =
  HttpClient.post client "/api/domain/{id}/dnssec" params

(** DELETE /api/domain/{id}/dnssec - Remove DNSSEC record *)
let delete_dnssec_record client params =
  HttpClient.delete client "/api/domain/{id}/dnssec" params

(** POST /api/domain/{id}/reg - Register domain nameserver *)
let register_nameserver client params =
  HttpClient.post client "/api/domain/{id}/reg" params

File: sdk/julia/src/dns_manage.jl
# DNS Manage: domain DNS records, DNSSEC, nameserver registration
struct DnsManageResource
    http::HttpClient
end

# GET /api/domain/{id}/dns - List DNS records
function get_records(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/domain/{id}/dns"; params = params)
end

# POST /api/domain/{id}/dns - Create DNS record
function create_record(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/domain/{id}/dns"; params = params)
end

# PUT /api/domain/{id}/dns/{index} - Update DNS record
function update_record(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_put(r.http, "/api/domain/{id}/dns/{index}"; params = params)
end

# DELETE /api/domain/{id}/dns/{index} - Remove DNS record
function delete_record(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_delete(r.http, "/api/domain/{id}/dns/{index}"; params = params)
end

# GET /api/domain/{id}/dns/types - List supported record types
function get_record_types(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/domain/{id}/dns/types"; params = params)
end

# GET /api/domain/{id}/dnssec/flags - Get available DNSSEC flags
function get_dnssec_flags(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/domain/{id}/dnssec/flags"; params = params)
end

# GET /api/domain/{id}/dnssec - Get DNSSEC records
function get_dnssec_records(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/domain/{id}/dnssec"; params = params)
end

# POST /api/domain/{id}/dnssec - Create DNSSEC record
function create_dnssec_record(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/domain/{id}/dnssec"; params = params)
end

# DELETE /api/domain/{id}/dnssec - Remove DNSSEC record
function delete_dnssec_record(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_delete(r.http, "/api/domain/{id}/dnssec"; params = params)
end

# POST /api/domain/{id}/reg - Register domain nameserver
function register_nameserver(r::DnsManageResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/domain/{id}/reg"; params = params)
end
